packages/sdk-core/src/modules/database/BacktraceDatabase.ts

```diff
@@ -0,0 +1,201 @@
+
+    /**
+     * Returns stored references to Backtrace records
+     * @returns available records in the database
+     */
+    public get(): BacktraceDatabaseRecord[] {
+        return this._databaseRecordContext.get();
+    }
+
+    /**
+     * @returns Returns number of records stored in the Database
+     */
+    public count(): number {
+        return this._databaseRecordContext.count();
+    }
+
+    /**
+     * Disables database integration. After running the dispose method, you cannot
+     * execute any database operations.
+     */
+    public dispose() {
+        this._enabled = false;
+        clearInterval(this._intervalId);
+    }
+
+    /**
+     * Removes the database record
+     * @param record database records
+     */
+    public async remove(record: BacktraceDatabaseRecord) {
+        if (!this._enabled) {
+            return;
+        }
+        this._databaseRecordContext.remove(record);
+        await this._storageProvider.delete(record);
+    }
+
+    /**
+     * Prepare database to insert records
+     * @param totalNumberOfRecords number of records to insert
+     */
+    private prepareDatabase(totalNumberOfRecords = 1) {
+        const numberOfRecords = this.count();
+        if (numberOfRecords + totalNumberOfRecords > this._maximumRecords) {
+            this._databaseRecordContext.dropOverflow(totalNumberOfRecords);
+        }
+    }
+
+    private async loadReports(): Promise<void> {
+        const records = await this._storageProvider.get();
+        if (records.length > this._maximumRecords) {
+            records.length = this._maximumRecords;
+        }
+        this.prepareDatabase(records.length);
+        this._databaseRecordContext.load(records);
+    }
+
+    private async setupDatabaseAutoSend() {
+        if (this._options?.autoSend === false) {
+            return;
+        }
+
+        const sendDatabaseReports = async () => {
+            await this.sendRecords();
+        };
+        this._intervalId = setInterval(sendDatabaseReports, this._retryInterval);
+        await this.sendRecords();
+    }
+
+    private async sendRecords() {
+        let submissionFailure = false;
+        for (let bucketIndex = 0; bucketIndex < this._databaseRecordContext.bucketCount; bucketIndex++) {
+            if (submissionFailure) {
+                break;
+            }
+            for (const record of this._databaseRecordContext.getBucket(bucketIndex)) {
+                if (record.locked) {
+                    continue;
+                }
+                try {
+                    record.locked = true;
+                    const result = await this._requestHandler.send(record.data, record.attachments);
+                    if (result.status === 'Ok') {
+                        this.remove(record);
+                        continue;
+                    }
+                    submissionFailure = true;
+                    this._databaseRecordContext.increaseBucket(bucketIndex);
+                    break;
```

Comment:
Instead of assigning to `submissionFailure` and checking it afterwards, you could just `return` here.